topic_monitor/scripts/data_publisher.py

```diff
@@ -0,0 +1,136 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import sys
+from time import sleep
+
+import rclpy
+from rclpy.qos import QoSDurabilityPolicy, QoSHistoryPolicy, QoSReliabilityPolicy
+from rclpy.qos import QoSProfile
+
+from std_msgs.msg import Header
+
+
+default_depth = 10
+
+
+def main():
+    parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument(
+        'data_name',
+        nargs='?',
+        default='topic1',
+        help='Name of the data (must comply with ROS topic rules)')
+
+    parser.add_argument(
+        '--best-effort',
+        action='store_true',
+        default=False,
+        help='Set QoS reliability option to "best effort"')
+
+    parser.add_argument(
+        '--transient-local',
+        action='store_true',
+        default=False,
+        help='Set QoS durability option to "transient local"')
+
+    parser.add_argument(
+        '--depth',
+        type=int,
+        default=default_depth,
+        action='store',
+        help='Size of the QoS history depth')
+
+    parser.add_argument(
+        '--keep-all',
+        action='store_true',
+        default=False,
+        help='Set QoS history option to "keep all" (unlimited depth, subject to resource limits)')
+
+    parser.add_argument(
+        '--payload-size',
+        type=int,
+        action='store',
+        default=0,
+        help='Size of data payload to send')
+
+    parser.add_argument(
+        '--period',
+        type=float,
+        default=0.5,
+        action='store',
+        help='Time in seconds between messages')
+
+    parser.add_argument(
+        '--end-after',
+        type=int,
+        action='store',
+        help='Script will exit after publishing this many messages')
+
+    args = parser.parse_args()
+
+    rclpy.init()
+
+    qos_profile = QoSProfile()
+
+    qos_profile.reliability = QoSReliabilityPolicy.RMW_QOS_POLICY_BEST_EFFORT if args.best_effort \
+        else QoSReliabilityPolicy.RMW_QOS_POLICY_RELIABLE
+
+    qos_profile.history = QoSHistoryPolicy.RMW_QOS_POLICY_KEEP_ALL_HISTORY if args.keep_all \
+        else QoSHistoryPolicy.RMW_QOS_POLICY_KEEP_LAST_HISTORY
+
+    qos_profile.depth = args.depth
+
+    qos_profile.durability = \
+        QoSDurabilityPolicy.RMW_QOS_POLICY_TRANSIENT_LOCAL_DURABILITY if args.transient_local \
+        else QoSDurabilityPolicy.RMW_QOS_POLICY_VOLATILE_DURABILITY
```

Comment:
if these end up being 3-line oneliners, maybe using the classic block syntax would be more readable